api/client.go

```diff
@@ -68,6 +69,9 @@ type Client struct {
 
 	// The logger used
 	logger logger.Logger
+
+	// server-specified HTTP request headers to include in all requests
+	requestHeaders http.Header
```

Comment:
Here's the internal state where the server-specified headers live inside the API client.